lib/real-device-management.js

```diff
@@ -67,20 +68,46 @@ async function runRealDeviceReset(device, opts) {
  * @param {InstallOptions} [opts]
  */
 async function installToRealDevice(device, app, bundleId, opts) {
-  if (!device.udid || !app) {
-    log.debug('No device id or app, not installing to real device.');
+  if (!device.udid || !app || !bundleId) {
```

Comment:
https://github.com/appium/appium-xcuitest-driver/pull/2054 must merge first since this check will make the current installation fail

Reply:
I have merged the above PR. Please rebase